crates/core/src/sql/execute.rs

```diff
@@ -187,31 +191,47 @@ pub struct SqlResult {
 
 /// Run the `SQL` string using the `auth` credentials
 pub async fn run(
-    db: &RelationalDB,
-    sql_text: &str,
+    db: Arc<RelationalDB>,
+    sql_text: String,
     auth: AuthCtx,
-    subs: Option<&ModuleSubscriptions>,
-    module: Option<&ModuleHost>,
+    subs: Option<ModuleSubscriptions>,
+    module: Option<ModuleHost>,
     head: &mut Vec<(Box<str>, AlgebraicType)>,
 ) -> Result<SqlResult, DBError> {
+    match module {
+        Some(module) => {
+            let info = module.info.clone();
+            module.call_view_sql(info, db, sql_text, auth, subs).await
+        }
+        None => run_from_module::<crate::host::wasmtime::WasmtimeInstance>(None, db, sql_text, auth, subs, head)
+            .map(|x| x.0),
+    }
+}
+
+pub fn run_from_module<I: WasmInstance>(
```

Comment:
Would be nice to have some commentary here about how & why this must not be `async`. Also, a comment about where this function is expected to run (on the database core? On the global executor, and it schedules something to the database core internally?) would be appreciated.